language/move-binary-format/src/unit_tests/compatibility_tests.rs

```diff
@@ -115,3 +204,30 @@ fn deprecated_add_script_visibility() {
         .check(&friend_module, &script_module)
         .is_err());
 }
+
+#[test]
+fn check_exact_and_unchange_same_module() {
```

Comment:
It would also be great to include some tests that show that the check is permutation invariant (for all the pools).